### PR TITLE
Add Mammalian Phenotype Ontology

### LINKED ISSUE #240: Add the Mammalian Phenotype Ontology
CAM-KP would like to support terms from the Mammalian Phenotype Ontology (e.g. https://www.informatics.jax.org/vocab/mp_ontology/MP:0000231).

### PR DESCRIPTION
**Pull request by gaurav**
This PR adds terms from the Mammalian Phenotype Ontology as requested by CAM-KP (#240). I added all the MP identifiers as well as mappings from https://github.com/mapping-commons/mh_mapping_initiative to connect it to HP and other identifiers. I added `sssom` as an explicit prerequisite so we can use it to read the SSSOM files in that GitHub repo, but that caused a lot of our other prerequisites to change (hence all the changes to requirements.lock).

We [end up with](https://github.com/user-attachments/files/16595759/PhenotypicFeature.txt) 13,335 cliques that consist only of an MP identifier and 588 cliques that combine MP identifiers with other identifiers. We have no cases where an `MP:` identifier is chosen over other identifiers; the clique leaders we generate are:

```
  26 MESH
8498 EFO
13335 MP
16014 HP
19843 NCIT
315066 UMLS
```

There are a bunch of mapping issues, such as:
* [MP:0001914 "hemorrhage"](https://www.informatics.jax.org/vocab/mp_ontology/MP:0001914) duplicates [NCIT:C26791 "hemorrhage"](https://nodenormalization-sri.renci.org/1.5/get_normalized_nodes?curie=NCIT%3AC26791&conflate=true&drug_chemical_conflate=false&description=false), which includes MESH and MEDRA identifiers.
  * We might be able to [use EFO](https://www.ebi.ac.uk/ols4/ontologies/EFO/classes/http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMP_0001914?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMP_0001914) to make this connection, but UberGraph doesn't [contain this edge](https://yasgui.triply.cc/#query=PREFIX%20dcterms%3A%20%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Fterms%2F%3E%0APREFIX%20obo%3A%20%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2F%3E%0APREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0Aprefix%20MP%3A%20%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMP_%3E%0ASELECT%20DISTINCT%20*%0AWHERE%20%7B%0A%20%20MP%3A0001914%20%3Fp%20%3Ft%0A%7D%0ALIMIT%201000&endpoint=https%3A%2F%2Fubergraph.apps.renci.org%2Fsparql&requestMethod=POST&tabTitle=Query%202&headers=%7B%7D&contentTypeConstruct=application%2Fn-triples%2C*%2F*%3Bq%3D0.9&contentTypeSelect=application%2Fsparql-results%2Bjson%2C*%2F*%3Bq%3D0.9&outputFormat=table&outputSettings=%7B%22pageSize%22%3A1000%7D) since UberGraph doesn't have EFO.
* [MP:0003342 "accessory spleen"](https://www.informatics.jax.org/vocab/mp_ontology/MP:0003342) is cliqued with [HP:0001748 "Polysplenia"](https://monarchinitiative.org/HP:0001748) instead of [HP:0001747 "Accessory spleen"](https://monarchinitiative.org/HP:0001747)

Ordinarily I would be nervous about including MP without more/better mappings, but since this isn't going to affect autocomplete (where we specifically filter to MONDO|HP) and that MP identifiers aren't (currently) clique leaders, I think we can merge this in now and then fix cliquing issues if anybody runs into issues with them (for now, probably only CAM-KP).

Closes #240.

WIP: see how often this new information merges cliques in ways we don't expect.

Should be merged after PR #365.

### CODE REVIEW

**Comment by gaurav** [comment]
- [ ] Message Nico and see if he knows of any mapping sources.
- [ ] Talk to Chris about maybe allowing the MP IDs in now and fixing them later, maybe as part of an exact-label cleanup.
- [ ] Give up and get rid of the MP-only cliques, so we only include MP identifiers if they combine with an existing clique.

**Comment by gaurav** [comment]
Superceded by PR #790. Closing.
